### PR TITLE
Completed Network Data structure

### PR DESCRIPTION
**Pull request by JethroWarnett**
- New edges are added to the edge array
- New nodes are added to the node array
- Deleted edges are removed from the edge array
- Deleted node are removed from the node array

### CODE REVIEW

File: src/app/components/fairChain/fairChain.component.ts [hunk 1]
```diff
@@ -121,18 +124,46 @@ export class FairChainComponent implements OnInit {
...
       if (this.isChangeNodeLabel) this.network.editNode();
     }
   }
 
+  public debugPrint() {
+    console.clear();
+    console.log(JSON.stringify(this.nodes));
+    console.log(JSON.stringify(this.edges));
+  }
+
```
Comment (GattoPianista) [inline src/app/components/fairChain/fairChain.component.ts:166]: I would not push it on dev

**Review by GattoPianista — APPROVED** [review]

**Review by LYCChiang — APPROVED** [review]

**Review by Korab-Bejta — APPROVED** [review]
ulooks good!

File: src/app/components/fairChain/fairChain.component.ts [hunk 2]
```diff
@@ -121,18 +124,46 @@ export class FairChainComponent implements OnInit {
...
+  //Delete Elements that have specific id
+  private deleteIdFromArray(arrayToTrim: vis.Node[]|vis.Edge[], ids: vis.IdType[]) {
+    ids.forEach(id => {
+      let index = this.IdToIndex(arrayToTrim, id)
+      arrayToTrim.splice(index, 1);
+    });
+  }
+
+  //TODO: Change datastructure if performance is bad
+  private IdToIndex(array: vis.Node[]|vis.Edge[], id: vis.IdType) {
```
Comment (mdeliso97) [inline src/app/components/fairChain/fairChain.component.ts:150]: IdTOIndex should be "idToIndex"

**Review by mdeliso97 — CHANGES_REQUESTED** [review]
The overall Implementation looks clear and very usefull, there are some corrections to be done. I think the correct position of the nodes/edges might be usefull if we want to implement drop down menus or "hided pop-ups" with options as discussed with the client (the pop ups should be located where the selected node/edge is).
There is a capital letter to correct (see corrections).
There are some comments missing on some methods.
Some Methods name are not self explained (such like debugPrint, deleteIdFromArray ->"id of what?", findIndex, etc.).
Some TODO are missleading: E.g. line 149: Change datastructure if performance is bad -> what do you mean here?
I couldn't figure out what is the debug button that you made

File: src/app/components/fairChain/fairChain.component.ts (hunk 1 above)
Comment (mdeliso97) [reply src/app/components/fairChain/fairChain.component.ts:166]: The overall implementation is merged in the changePageLayout

**Pull request closed without merging**